File: cli-tools/automation/python/remove_background.py
import argparse
import requests
import json
import sys
from utils import BASE_URL, get_access_token

def remove_background(image_url):
    try:
        token = get_access_token()
        headers = {
            'Authorization': f'Bearer {token}',
            'Content-Type': 'application/json'
        }
        payload = {
            'image_url': image_url,
            'transparent_background': True
        }
        
        response = requests.post(f"{BASE_URL}/images/remove-background", json=payload, headers=headers)
        response.raise_for_status()
        
        print(json.dumps(response.json(), indent=2))
        
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='Remove background using Pixelz Automation API')
    parser.add_argument('--url', required=True, help='Image URL')
    args = parser.parse_args()
    
    remove_background(args.url)

File: cli-tools/automation/node/remove-background.js
const axios = require('axios');
const { BASE_URL, getAccessToken } = require('./utils');

const args = process.argv.slice(2);
const imageUrl = args.find(arg => arg.startsWith('--url='))?.split('=')[1];

if (!imageUrl) {
    console.log("Usage: node remove-background.js --url=<image_url>");
    process.exit(1);
}

async function removeBackground() {
    try {
        const token = await getAccessToken();
        const payload = {
            image_url: imageUrl,
            transparent_background: true
        };

        const response = await axios.post(`${BASE_URL}/images/remove-background`, payload, {
            headers: {
                'Authorization': `Bearer ${token}`,
                'Content-Type': 'application/json'
            }
        });

        console.log(JSON.stringify(response.data, null, 2));

    } catch (error) {
        console.error("Error:", error.message);
        if (error.response) console.error(error.response.data);
    }
}

removeBackground();
